### ISSUE TITLE
nuget.exe should skip DNX projects

### ISSUE THREAD

**Issue by emgarten** [NuGet/Home#1227]
nuget.exe will incorrectly restore project.json files from DNX projects when a solution contains .xproj files or nuget.exe restore is called directly on the project.json file.

DNX projects should be skipped when restoring a solution. In scenarios where the user runs nuget.exe restore on an .xproj or project.json file NuGet should error out and display a message letting users know that nuget.exe does not support this.

**Comment by yishaigalatzer** [comment]
It should probably not error out, and rather display a warning.

**Comment by yishaigalatzer** [comment]
We want to fix this immediately as it breaks VSO.

We want not to error, but skip.

Scenario:

It tries to find the `project.json` **after** the `.xproject` instead of combining the folder of the project and `project.json`

```
"C:\VSO Build\nuget.exe" restore "C:\VSO Build\_work\e8bff5de\MyProject\MyProject.Server.sln"
MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'.
Could not find a part of the path 'C:\VSO Build\_work\e8bff5de\MyProject\MyProject.Service.Web\MyProject.Service.Web.xproj\project.json'.
```

This is a pretty serious issue, as it completely fails any VSO Build with Package Restore enabled on the Solution. 2.8.x just ignored the project files ...

**Comment by Mertsch** [comment]
Yes, just disable it, like it was before 3.2 (VSO build uses 3.2 as soon as you update the VS extension) then you can finally restore packages that require 3.x (like System.Web.Http 4.0.1 beta), but you can't complete your build of the solution.

My current workaround is ... split everything C# into one solution with VS package restore and
everything DNX into another solution without package restore (restores with separate .ps1)

**Comment by emgarten** [comment]
NuGet/NuGet.PackageManagement@4c04f8955b5523014a6d28c99ff7073e80b1424a

**Comment by zhili1208** [comment]
verified this, it's fixed in 3.2.1
